core/src/main/java/org/apache/iceberg/BaseReplacePartitions.java

```diff
@@ -79,23 +79,32 @@ public ReplacePartitions validateNoConflictingData() {
     return this;
   }
 
+  @Override
+  public BaseReplacePartitions toBranch(String branch) {
+    targetBranch(branch);
+    return this;
+  }
+
   @Override
   public void validate(TableMetadata currentMetadata, Snapshot snapshot) {
     if (validateConflictingData) {
       if (dataSpec().isUnpartitioned()) {
-        validateAddedDataFiles(currentMetadata, startingSnapshotId, Expressions.alwaysTrue());
+        validateAddedDataFiles(
+            currentMetadata, startingSnapshotId, Expressions.alwaysTrue(), snapshot);
       } else {
-        validateAddedDataFiles(currentMetadata, startingSnapshotId, replacedPartitions);
+        validateAddedDataFiles(currentMetadata, startingSnapshotId, replacedPartitions, snapshot);
```

Comment:
Minor: I think it makes more sense to use `parent` rather than `snapshot` for the parent snapshot variable. Not a blocker, but it makes it easier to understand what's going on.